### ISSUE TITLE
Contacts can have follow-up cases but remain with the status 'Active contact'

### ISSUE THREAD

**Issue by adinaflorea9** [SORMAS-Foundation/SORMAS-Project#12742]
### Bug Description

Follow-up issue for #8337.

Cases can be created for contacts and the status of the contact remains unconfirmed both in the web and mobile app. In the web app, the user can not proceed to create the case unless they first save the changing of the contact status.

### Reproduction Steps

1. Log in the mobile app with a Default Surveillance Officer;
2. Create a new contact or open an existing one that is not yet a 'Confirmed contact;;
3. Click to edit the 'Contact status' and change it to 'Confirmed contact';
4. Then scroll down and click on 'Create a case for this contact person';
5. Fill in the mandatory information in the 'Create new case' form and save;
6. Trigger a synchronization of the mobile app with the web app.
Actual behavior: Both in the web app and in the mobile app, the contact is still 'Active contact' as after step 3 the user did not save the changes and at step 4 the user was not asked to first save the changes they made (similar to how it is done in the web app where the user is asked to).

### Expected Behavior

The user should be informed that they need to save the contact after updating the contact status and before proceeding to create the case for the contact person. Also make sure that the user is not be able to click to create the case if the changes on the contact were first not saved.

### Affected Area(s)

Android app

### SORMAS Version

1.91.0

### Android version/Browser

Android Studio Emulator

### User Role(s)

Surveillance Officerf

### Server URL

local machine

**Comment by carina29** [comment]
@adinaflorea9 @MateStrysewske I've done the changes requested: when the create case activity will be started, the contact classification of the saved contact will be checked (not the edited contact) and if this classification is confirmed, the redirecting to the new case creation will be done.
Otherwise, a warning will be shown as it was requested.

During testing I 've found some scenarios for which I'm not sure how should I proceed because in the webapp we have a "Discard or save changes pop-up" but in the mobile app we do not have this implemented.
**In the web app if the contact's classification is :**
1) **Not a Contact** -> is not appearing the "Create a case" button even after editing the classification ( without saving)
2) **Unconfirmed** -> "Create a case" button is appearing even if the contact classification will be changed to something else and in all the scenarios a popup will appear with this message: "You can only convert contacts to cases that have been confirmed. Please confirm this contact before creating a case for its contact person."
3) **Confirmed** -> after changes without saving (for example changing of the contact's classification), "Unsaved changes" pop-up will appear

Right now (after implementing the requested changes in this ticket), we will check all the time the saved contact's classification and based on that we display or not a warning message. ( even if for example the saved contact is confirmed but the edited one is unconfirmed or any other combination).

Is this ok? 
We do not have yet a "Save or discard unsaved changes" option in Android.

**Comment by MateStrysewske** [comment]
I think that's fine, yes.

**Comment by carina29** [comment]
We clarified with @Adina my questions from above.
The implementation will remain in the same way as it was requested: 
**The saved contact classification will be checked if is confirmed or not** and based on this, the following scenarios will exist:

1) "Unconfirmed" contact changed to "Confirmed" or "Not a Contact" without saving -> "Create new case" button appears but after clicking it the warning message will appear:
![image](https://github.com/SORMAS-Foundation/SORMAS-Project/assets/47103965/171ca70b-e503-4e91-ae84-aa06572e0809)

2) "Confirmed" changed to "Not a contact" or "Unconfirmed" -> "Create new case" button appears and the new case creation will be possible after clicking the button

3) "Not a contact" changed to "Confirmed" or "Unconfirmed" -> the "Create new case" button will not appear

**Comment by adinaflorea9** [comment]
Verified ticket on the local machine using the latest version of SORMAS from the development branch - 1.92.0-SNAPSHOT(3f91bca).
